master/internal/stream/system_intg_test.go

```diff
@@ -0,0 +1,200 @@
+//go:build integration
+// +build integration
+
+package stream
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/labstack/echo/v4"
+	"github.com/stretchr/testify/require"
+
+	detContext "github.com/determined-ai/determined/master/internal/context"
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/stream"
+	"github.com/determined-ai/determined/master/test/streamdata"
+)
+
+func simpleUpsert(i stream.PreparableMessage) interface{} {
+	return i
+}
+
+func recordDeletion(s1, s2 string) interface{} {
+	fmt.Println("recording deletion:", s1+" and "+s2)
+	return s2
+}
+
+type startupReadWriter struct {
+	data           []interface{}
+	startupMessage *StartupMsg
+}
+
+func (t *startupReadWriter) ReadJSON(data interface{}) error {
+	if t.startupMessage == nil {
+		return fmt.Errorf("startup message has been sent")
+	}
+	targetMsg, ok := data.(*StartupMsg)
+	if !ok {
+		return fmt.Errorf("target message type is not a pointer to StartupMsg")
+	}
+	targetMsg.Known = t.startupMessage.Known
+	targetMsg.Subscribe = t.startupMessage.Subscribe
+	t.startupMessage = nil
+	return nil
+}
+
+func (t *startupReadWriter) Write(data interface{}) error {
+	t.data = append(t.data, data)
+	return nil
+}
+
+func (t *startupReadWriter) Close() error {
+	return nil
+}
+
+func TestReadWriter(t *testing.T) {
```

Comment:
What code is this test testing?  I can't quite tell

Reply:
it's slightly misnamed, but this is testing the `startupReadWriter` struct, primarily to make sure the `ReadJSON` works in the capacity I need it to